Fix Tooltip "toggle" bug (#873)
CrowdStrike/glide-core: .changeset/ten-monkeys-rhyme.md

---
'@crowdstrike/glide-core': patch
---

> - Button no longer emits a "toggle" event on hover when not disabled.

Reviewer: I'm confused about the "when not disabled" part of this. What did the disabled state of a button have to do with the toggle event from the tooltip?

Author: Button's only supposed to show a Tooltip (and thus emit a "toggle" event on hover) when Button is disabled. Previously it was emitting a "toggle" event on hover when enabled.

Reviewer: > Button's only supposed to show a Tooltip (and thus emit a "toggle" event on hover) when Button is disabled.

But that's not automatic though, is it? From testing in `main`, if you just set a Button to disabled, the Tooltip is still disabled, but the "toggle" event is being emitted. Which I'd say is still wrong.

Might I suggest this wording?:

```suggestion
- Button no longer emits a "toggle" event on hover, unless it has a Tooltip enabled.
```

Author: That's not quite accurate. Button should never emit a "toggle" event on hover when enabled because its Tooltip is only shown when Button is disabled.

How about something like this?

```suggestion
- Button's `tooltip` attribute shows the tooltip only when Button is disabled. The tooltip's "toggle" event was previously emitted on hover even when Button wasn't disabled. "toggle" is now only emitted when Button is disabled.
```

Author: Beautiful.
